db/upgrade-X.X.X-X.Y.Z.sql

```diff
@@ -51,8 +51,127 @@ DELIMITER ;
 call ValidateVersion;
 DROP PROCEDURE IF EXISTS ValidateVersion;
 
+--
+-- Table structure for table `locationlog_history`
+--
+
+CREATE TABLE IF NOT EXISTS `locationlog_history` (
+  `id` int NOT NULL PRIMARY KEY AUTO_INCREMENT,
+  `tenant_id` int NOT NULL DEFAULT 1,
+  `mac` varchar(17) default NULL,
+  `switch` varchar(17) NOT NULL default '',
+  `port` varchar(20) NOT NULL default '',
+  `vlan` varchar(50) default NULL,
+  `role` varchar(255) default NULL,
+  `connection_type` varchar(50) NOT NULL default '',
+  `connection_sub_type` varchar(50) default NULL,
+  `dot1x_username` varchar(255) NOT NULL default '',
+  `ssid` varchar(32) NOT NULL default '',
+  `start_time` datetime NOT NULL default '0000-00-00 00:00:00',
+  `end_time` datetime NOT NULL default '0000-00-00 00:00:00',
+  `switch_ip` varchar(17) DEFAULT NULL,
+  `switch_mac` varchar(17) DEFAULT NULL,
+  `stripped_user_name` varchar (255) DEFAULT NULL,
+  `realm`  varchar (255) DEFAULT NULL,
+  `session_id` VARCHAR(255) DEFAULT NULL,
+  `ifDesc` VARCHAR(255) DEFAULT NULL,
+  `voip` enum('no','yes') NOT NULL DEFAULT 'no',
+  KEY `locationlog_view_mac` (`tenant_id`, `mac`, `end_time`),
+  KEY `locationlog_end_time` ( `end_time`),
+  KEY `locationlog_view_switchport` (`switch`,`port`,`end_time`,`vlan`),
+  KEY `locationlog_ssid` (`ssid`),
+  KEY `locationlog_session_id_end_time` (`session_id`, `end_time`)
+) ENGINE=InnoDB;
+
+INSERT INTO locationlog_history (
+ `tenant_id`, `mac`, `switch`, `port`, `vlan`, `role`, `connection_type`, 
+ `connection_sub_type`, `dot1x_username`, `ssid`, `start_time`, `end_time`, `switch_ip`, 
+ `switch_mac`, `stripped_user_name`, `realm`, `session_id`, `ifDesc`, `voip`)  
+    SELECT `tenant_id`, `mac`, `switch`, `port`, `vlan`, `role`, `connection_type`, 
+    `connection_sub_type`, `dot1x_username`, `ssid`, `start_time`, `end_time`, `switch_ip`,
+    `switch_mac`, `stripped_user_name`, `realm`, `session_id`, `ifDesc`, `voip` 
+    FROM locationlog WHERE end_time != '0000-00-00 00:00:00';
+
+DELETE FROM locationlog 
```

Comment:
As discussed, it would probably be better to rename the current locationlog to locationlog_history and then populate a new locationlog table with the latest active entry